topic-controller/src/main/java/io/strimzi/controller/topic/Controller.java

```diff
@@ -334,30 +334,48 @@ public Controller(Vertx vertx, Kafka kafka,
-                    reconcile(cm, k8sTopic, kafkaTopic, privateTopic, reconcileResult -> {
+        Handler<Future<Void>> futureHandler = new Reconciliation("reconcile") {
+            @Override
+            public void handle(Future<Void> fut) {
+
+                try {
+                    Topic k8sTopic = cm != null ? TopicSerialization.fromConfigMap(cm) : null;
+                    Future<Topic> topicResult = Future.future();
+                    Future<TopicMetadata> metadataResult = Future.future();
+                    kafka.topicMetadata(topicName, metadataResult.completer());
+                    topicStore.read(topicName, topicResult.completer());
+                    CompositeFuture.all(topicResult, metadataResult).setHandler(ar -> {
+
+                        if (ar.succeeded()) {
+                            Topic privateTopic = ar.result().resultAt(0);
+                            TopicMetadata kafkaTopicMeta = ar.result().resultAt(1);
+                            Topic kafkaTopic = TopicSerialization.fromTopicMetadata(kafkaTopicMeta);
+                            reconcile(cm, k8sTopic, kafkaTopic, privateTopic, reconcileResult -> {
+                                if (reconcileResult.succeeded()) {
+                                    LOGGER.info("Success reconciling ConfigMap {}: ", logConfigMap(cm));
+                                    fut.complete();
+                                } else {
+                                    LOGGER.error("Error reconciling ConfigMap {}: ", logConfigMap(cm), reconcileResult.cause());
+                                    fut.fail(reconcileResult.cause());
+                                }
+                            });
+                        } else {
+                            LOGGER.error("Error reconciling ConfigMap {}: ", logConfigMap(cm), ar.cause());
+                            fut.fail(ar.cause());
+                        }
                     });
-                } else {
-                    LOGGER.error("Error reconciling ConfigMap {}: ", cm.getMetadata().getName(), ar.cause());
+                } catch (InvalidConfigMapException e) {
+                    LOGGER.error("Error reconciling ConfigMap {}: Invalid 'data' section: ", logConfigMap(cm), e.getMessage());
+                    fut.fail(e);
+                } catch (ControllerException e) {
+                    LOGGER.error("Error reconciling ConfigMap {}: ", logConfigMap(cm), e);
+                    fut.fail(e);
                 }
-            });
-        } catch (InvalidConfigMapException e) {
-            LOGGER.error("Error reconciling ConfigMap {}: Invalid 'data' section: ", cm.getMetadata().getName(), e.getMessage());
-        } catch (ControllerException e) {
-            LOGGER.error("Error reconciling ConfigMap {}: ", cm.getMetadata().getName(), e);
-        }
+            }
+        };
+        inFlight.enqueue(topicName, resultHandler, futureHandler);
```

Comment:
I think we should rename `futureHandler` → `action` or something to make clear that it's the piece of work to be run. I think we should swap the order of the parameters in `enqueue()`, so the action goes before the result handler. That would make it more consistent with the Vertx APIs, like `executeBlocking()`.

Reply:
Agree, do we want do that in this PR ? Maybe a different one involving the other methods could be better, wdyt ?

Reply:
I think it's a trivial refactoring in the IDE, so we can just do it here.

Reply:
Yes that's true. Just done.